### PR TITLE
Seeking Review: Documentation of the feature.

### PR DESCRIPTION
**Pull request by edburns**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+This module interfaces with the TLS implementation of Oracle JDK8 to
+provide an implementation of ALPN.  The module was written for use in
+Grizzly, but is intended as a general purpose solution to allow JDK8
+based software to use ALPN.
+
+---
+
+## How to Avoid IP Taint When Examining this Code
```
Comment (ghost) [inline README.md]: I don't think this is necessary.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+implementation will insert one or more `Filter` implementations into the
+`FilterChain` which is used to process HTTP requests and cause HTTP
+responses to be sent.
+
+Grizzly itself uses this `AddOn` concept to provide HTTP/2 support, in
+the form of
+[`Http2AddOn`](https://github.com/javaee/grizzly/blob/e0e9200479851078d9cf2bad1cf29fa72f525437/modules/http2/src/main/java/org/glassfish/grizzly/http2/Http2AddOn.java)
+This `AddOn` implementation inserts some filters in the chain in the
+proper order to ensure that ALPN concerns are handled first, and then
+the HTTP/2 protocol is handled.  The former is of interest here.
```
Comment (ghost) [inline README.md]: Better phrased as something like:  in addition to registering filters for HTTP/2 processing, it registers a callback with the ALPN extension.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    * Call `NegotiationSupport.addNegotiator` passing the `SSLEngine`
+    and the client negotiator impl.
+
+    If `getUseClientmode` is false,
+
+    * do the same as above, but use the server negotiator impl.
+
+    By the time `onStart` is invoked, the handshake listener will have
+    been initialized with implementations of `AlpnClientNegotiator` and
+    `AlpnServerNegotiator`.
```
Comment (ghost) [inline README.md]: via the Http2Addon

File: README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    By the time `onStart` is invoked, the handshake listener will have
+    been initialized with implementations of `AlpnClientNegotiator` and
+    `AlpnServerNegotiator`.  
+
+## Deployment Configuration
+
+The above programmatic configuration steps will make it so the simple
+act of including the module built from the `bootstrap` sub-module of
+this repository in the "endorsed" directory of the JVM will enable ALPN
+support in that JVM.
```
Comment (ghost) [inline README.md]: I would recommend documenting the use of bootclasspath vs endorsed of the JVM so as to not impact all applications that are using the JVM, only those that need ALPN.

File: README.md (hunk 1 above)
Comment (edburns) [reply README.md]: Removed.

File: README.md (hunk 2 above)
Comment (edburns) [reply README.md]: Done.

File: README.md (hunk 3 above)
Comment (edburns) [reply README.md]: Done.

File: README.md (hunk 4 above)
Comment (edburns) [reply README.md]: Done.

**Comment by edburns** [comment]
@rlubke Please review and accept this PR if you agree.  Once you do, I'll point the customer to it.

Thanks,

Ed

**Pull request merged**